Common/TableProducer/PID/pidTPC.cxx

```diff
@@ -163,33 +164,23 @@ struct tpcPid {
       } else {
         if (networkSetNumThreads > 0) {
           LOGP(info, "Not running on Hyperloop. Threads for neural network inference are fixed: {} threads", std::to_string(networkSetNumThreads));
-          activeThreads = networkSetNumThreads.value;
         }
       }
 
       /// CCDB and auto-fetching
-      ccdbApi.init(url);
+      network.setCcdbUrl(url);
       if (!autofetchNetworks) {
         if (ccdbTimestamp > 0) {
           /// Fetching network for specific timestamp
           LOG(info) << "Fetching network for timestamp: " << ccdbTimestamp.value;
-          std::map<std::string, std::string> metadata;
-          bool retrieve_success = ccdbApi.retrieveBlob(networkPathCCDB.value, ".", metadata, ccdbTimestamp.value, false, networkPathLocally.value);
-          if (retrieve_success) {
-            std::map<std::string, std::string> headers = ccdbApi.retrieveHeaders(networkPathCCDB.value, metadata, ccdbTimestamp.value);
-            if (headers.count("Valid-From") == 0) {
-              LOG(fatal) << "Valid-From not found in metadata";
+          bool fetchSuccess = network.fetchFromCCDB(networkPathCCDB, ccdbTimestamp.value, networkPathLocally, true);
+          if (fetchSuccess) {
+            if (enableNetworkOptimizations) {
+              Ort::SessionOptions* sessOpts = network.getSessionOptions();
+              sessOpts->SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_EXTENDED);
```

Comment:
I would add a setter in the new class and avoid to expose the pointer

Reply:
I did this explicitly to have the full flexibility of the ONNX options available for the user. But maybe you have an idea how this could be done with a function/class which still maintains the flexibility?

Reply:
Maybe we could expose some setters to enable specific options (btw this can be done at a later stage), not sure that having full flexibility is a good idea.